internal/service/advancedcluster/resource_advanced_cluster_test.go

```diff
@@ -14,6 +14,13 @@ import (
 	"go.mongodb.org/atlas-sdk/v20231115007/admin"
 )
 
+func TestMain(m *testing.M) {
```

Comment:
This PR is really hard to review 😢  Just an FYI that people in cloud were discussing this interesting blog post [The ideal PR is 50 lines long](https://graphite.dev/blog/the-ideal-pr-is-50-lines-long)

Reply:
thx for the comment. Sorry, I know and I normally try to do small PRs. In this case I try to explain in the PR description why I chose to go for a bigger PR here.